driver/src/price_estimation/dexag/api.rs

```diff
@@ -0,0 +1,183 @@
+#![allow(dead_code)]
+
+use anyhow::{anyhow, Context, Result};
+use ethcontract::Address;
+use isahc::prelude::*;
+use serde::{de::Error as _, Deserialize, Deserializer};
+use serde_with::rust::display_fromstr;
+use std::time::Duration;
+use url::Url;
+
+#[cfg_attr(test, mockall::automock)]
+/// Parts of the dex.ag api.
+pub trait DexagApi {
+    /// https://docs.dex.ag/api/tokens
+    fn token_list_full(&self) -> Result<Vec<Token>>;
```

Comment:
ubernit: maybe call this `get_token_list` and `get_price`? (I don't see a specific meaning of *full list* as there seems to be no other type of token list to query)